File: 2022.01.19/javascript/9020.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');
const T = Number(input[0]);
const prime = new Array(10001).fill(true);

for (let i = 2,j ; i<parseInt(Math.sqrt(10000))+1 ; i++){
    j = 2;
    while (i*j <= 10000){
        prime[i*j] = false;
        j++;
    }
}
for (let i = 1,n,a,b ; i<input.length-1 ; i++){
    n = Number(input[i]);
    a = parseInt(n/2);
    for ( ; a>=2 ; a--){
        b = n - a;
        if (prime[a] && prime[b]) break
    }
    console.log(a,b)
}

File: 2022.01.19/python/9020.py
prime = [True for _ in range(10001)]
prime[0] = prime[1] = False
for i in range(2,int((10000*2)**0.5)+1):
    j = 2
    while i*j <= 10000:
        prime[i*j] = False
        j += 1
for _ in range(int(input())):
    n = int(input())
    a = n//2
    while a:
        b = n-a
        if prime[a] and prime[b] : break
        a -= 1
    print(a,b)